soh/src/code/z_play.c

```diff
@@ -1087,6 +1087,14 @@ void Play_Update(PlayState* play) {
                 // Gameplay stat tracking
                 if (!gSaveContext.sohStats.gameComplete) {
                       gSaveContext.sohStats.playTimer++;
+                      // Use ITEM_KEY_BOSS for Ganon's boss key
+                      if ((gSaveContext.inventory.dungeonItems[10] & 1) &&
+                          gSaveContext.sohStats.timestamp[ITEM_KEY_BOSS] == 0) {
+                          gSaveContext.sohStats.timestamp[ITEM_KEY_BOSS] = GAMEPLAYSTAT_TOTAL_TIME;
+                      }
```

Comment:
see earlier comment about moving this to `GameplayStats_SetTimestamp`

Reply:
This was just the single "catch-all" spot that worked best for all rando cases + vanilla (see other comment).

Reply:
Makes sense to me!